src/liseq.py

```diff
@@ -31,4 +31,18 @@ def liseq(x: Sequence[Any]) -> list[int]:
     """
     best: list[int] = []
     # Explore all alternatives
+    for i in range(len(x)):
+        for j in range(i, len(x)):
```

Comment:
You can start at `i + 1` if you make `j` point one-past the end of the current range. You extend it when it is larger than the previous one (and there will always be a previous one if you start one past `i`).